envsec/internal/auth/user.go

```diff
@@ -84,10 +85,17 @@ func (a *Authenticator) parseToken(stringToken string) (*jwt.Token, error) {
 		return nil, errors.WithStack(err)
 	}
 
-	token, err := jwt.Parse(stringToken, jwks.Keyfunc)
+	accessToken, err := jwt.Parse(tokens.AccessToken, jwks.Keyfunc)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+	idToken, err := jwt.Parse(tokens.IDToken, jwks.Keyfunc)
 	if err != nil {
 		return nil, errors.WithStack(err)
 	}
 
-	return token, nil
+	return &User{
+		AccessToken: accessToken,
```

Comment:
To get a working accessToken as a string, you need to reference `User.AccessToken.Raw` because of that, can we make this a private field and make a method named `(u *User) AccessToken()` that returns `u.accessToken.Raw`?

Reply:
I kinda prefer using *jwt.Token over strings. 

For example awsfed uses *jwt.Token param which is safer than using a string (strings are more common so there's less compile time type enforcement)

Similarly `jetcloud` currently uses `user` because it is using the temporary `OrgID()` method, but in reality it should use just a token.